### DISCUSSION TITLE
Deployment issue: Host key verification failed. fatal: Could not read from remote repository.

### DISCUSSION THREAD

**Discussion by bhavc** [vercel/community#1597] [Help]
### Summary

Hi, been trying to deploy my next.js app for the past couple of days but ive been getting the same issue. 
Every time i try to deploy my app, i get the following: 
`
Host key verification failed.
--
20:52:57.004 | fatal: Could not read from remote repository.
20:52:57.004 |  
20:52:57.004 | Please make sure you have the correct access rights
20:52:57.004 | and the repository exists.
`

Ive tried everything from deleting and recreating the project, deleting the recreating the repo, uninstalling vercel from github and updating my ssh key. Nothing seems to work here. Not sure how to move forward from here

**Comment by amyegan** [comment]
Hi @bhavc! I have a few questions to help us find the source of the problem:
- Is this a new project, or one that was previous deployed successfully?
- Are you deploying with the standard [Vercel for GitHub](https://vercel.com/docs/concepts/deployments/git/vercel-for-github) integration, from local [CLI](https://vercel.com/docs/cli), or another method?
- If you go to your [GitHub Integrations settings](https://github.com/settings/installations) and click the "Configure" button next to the Vercel integration, can you verify that Vercel has access to your repo? You should have either "All repositories" selected or "Only select repositories" with your repo appearing in the listed of selected repositories
- If this error is related to a private dependency, have you followed the guide for [using private dependencies with Vercel](https://vercel.com/guides/using-private-dependencies-with-vercel)?

**Reply by bhavc** [reply]
Hey Amy. 
This is a new project but ive deployed another project (sveltekit) the same way as before without issue. My current project has not been deployed yet. 
Ive tried both the Vercel GIthub integration as well as the CLI 
I have given access to all repos. 
I do not believe this is a private dependency or has anything to do with a private dependency. 

To add, ive changed my githubs primary email address recently and i dont know if that could have affected this. 

Thanks

**Reply by bhavc** [reply]
I've figured out my issue. For some reason PNPM was the reason for this error. I changed my package manager to yarn and it seemed to work fine. Any idea why this is the case?
